ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -62,6 +77,43 @@
     return optional(result);
   }
 
+  private static <T> Optional<T> coerceArray(final Object value, final SqlArray targetType) {
```

Comment:
This method seems to expect a List as value, so shouldn't we type value as List instead of Object for better type safety? Also return type seems to be List too.

Reply:
Also type param doesn't seem to be doing anything useful unless I am missing something.

Reply:
I'll type it as a list and remove the type parameter.

Reply:
I am actually keeping the signature without specifying `List` anywhere. If I add `List` to the parameter, then I need to cast it and check mismatch in `doCoerce`, which will cause the number of ifs to grow beyond what checkstyle likes. If I have it in the return, then I'll just need to cast it again when returning it and since it's a private helper method we don't gain anything by having the extra type safety.